lighthouse-core/report/formatter.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+
+// Auto convert partial filenames to lookup for audits to use.
+const formatters = {};
+fs.readdirSync(__dirname + '/partials')
+  .filter(filename => filename.endsWith('.html'))
+  .forEach(filename => {
+    const baseName = filename.replace(/\.html$/, '');
```

Comment:
> so the value of the formatter property is the same as the name of the partial

(and that's now checked by [a test](https://github.com/GoogleChrome/lighthouse/pull/1838/files#diff-dd8576d3d840580d95a4a90f37ab89a2R31))